File: Dart_Source_Files/payload/QuestionList.dart
// ignore_for_file: file_names

class QuestionList {
  int scorePerQuestion;
  int gameVersion;
  List<int> questionCodes;

  QuestionList(
      {required this.scorePerQuestion,
      required this.gameVersion,
      required this.questionCodes});

  factory QuestionList.fromJson(Map<String, dynamic> json) {
    return QuestionList(
        scorePerQuestion: json["scorePerQuestion"],
        gameVersion: json["gameVersion"],
        questionCodes: List<int>.from(json["questionCodes"]));
  }

  @override
  String toString() {
    return "[Score Per Question: $scorePerQuestion, Game Version: $gameVersion, Question Codes: $questionCodes]";
  }
}

File: Dart_Source_Files/payload/QuestionList.js
class QuestionList {
  
    /**
     * 
     * @param {number} scorePerQuestion 
     * @param {number} gameVersion 
     * @param {number[]} questionCodes 
     */
    constructor(scorePerQuestion, gameVersion, questionCodes) {

        this.scorePerQuestion = scorePerQuestion;
        this.gameVersion = gameVersion;
        this.questionCodes = questionCodes;
    }

    /**
     * 
     * @param {object} json 
     * @returns {QuestionList}
     */
    static fromJson(json) {

        return new QuestionList(json.scorePerQuestion, json.gameVersion, json.questionCodes);
    }

    /**
     * 
     * @returns {string}
     */
    toString() {

        return `[Score Per Question: ${this.scorePerQuestion}, Game Version: ${this.gameVersion}, Question Codes: ${this.questionCodes}]`;
    }
}

export {QuestionList};